src/main/java/dev/morling/onebrc/WeatherStation.java

```diff
@@ -0,0 +1,518 @@
+/*
+ *  Copyright 2023 The original authors
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package dev.morling.onebrc;
+
+import java.util.Arrays;
+import java.util.Random;
+import java.util.List;
+
+import org.rschwietzke.CheaperCharBuffer;
+import org.rschwietzke.FastRandom;
+
+public class WeatherStation {
+    final static char[] NUMBERS = { '0', '1', '2', '3', '4', '5', '6', '7', '8', '9' };
+
+    public final String id;
+    public final double meanTemperature;
+
+    private final char[] firstPart;
+    private final Random random;
+    private final FastRandom fastRandom;
+
+    public WeatherStation(long seed, String id, double meanTemperature) {
+        this.id = id;
+        this.meanTemperature = meanTemperature;
+
+        long rngSeed = ((long) id.hashCode()) ^ seed;
+
+        this.random = new Random(rngSeed);
+        this.fastRandom = new FastRandom(rngSeed);
+
+        this.firstPart = (id + ";").toCharArray();
+    }
+
+    public double measurement() {
+        double m = this.random.nextGaussian(this.meanTemperature, 10);
+        return Math.round(m * 10.0) / 10.0;
+    }
+
+    /**
+     * We write out data into the buffer to avoid string conversion
+     * We also no longer use double and gaussian, because for our
+     * purpose, the fake numbers here will do it. Less
+     *
+     * @param buffer the buffer to append to
+     */
+    public void measurement(final CheaperCharBuffer buffer) {
+
+        // fake -10.9 to +10.9 variance without double operations and rounding
```

Comment:
Ah, okay: using `java.util.Random` instead of `FastRandom` causes this path to be "only" 1.5x faster, as opposed to about 3.5x faster (from my limited measurement)